### PR TITLE
Fix space issues for database backup

### PR DESCRIPTION
**Pull request by RMcVelia**
### Context

Database backup/restore still failing with space issues after adding compression
It's currently failing during the local creatdb as there is only 7G available

Filesystem     1K-blocks     Used Available Use% Mounted on
/dev/root       87204404 79796528   7391492  92% /

### Changes proposed in this pull request

- Split steps to make it clearer where it's failing
- add step to remove unused files from the github runner

### Guidance to review

check code
see run https://github.com/DFE-Digital/register-trainee-teachers/actions/runs/6411121899/job/17405878447

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: .github/workflows/database-backup.yml [hunk 1]
```diff
@@ -105,20 +105,41 @@ jobs:
...
+        DATABASE_NAME: register_trainee_teachers
+        PGUSER:  postgres
+        PGPASSWORD: postgres
+        PGHOST: localhost
+        PGPORT: 5432
 
-        echo "::group::Backup Sanitised Database"
+    - name: Dump the Sanitised Database
+      run: |
         pg_dump --encoding utf8 --compress=1 --clean --no-owner --if-exists -d ${DATABASE_NAME} -f backup_sanitised.sql.gz
```
Comment (defong) [inline .github/workflows/database-backup.yml:142]: Maybe compress some more?

```suggestion
        pg_dump --encoding utf8 --compress=9 --clean --no-owner --if-exists -d ${DATABASE_NAME} -f backup_sanitised.sql.gz
```
With 9 being the max

or 
```suggestion
        pg_dump --encoding utf8 --compress=6 --clean --no-owner --if-exists -d ${DATABASE_NAME} -f backup_sanitised.sql.gz
```

With 6 being the underlaying default

File: .github/workflows/database-backup.yml [hunk 2]
```diff
@@ -105,20 +105,41 @@ jobs:
       env:
         SQLPAD_USER: ${{ secrets.SQLPAD_USER }}
 
-    - name: Sanitise the Database backup
+    - name: Disk cleanup
```
Comment (defong) [inline .github/workflows/database-backup.yml:108]: Can we not resize the machine?

File: .github/workflows/database-backup.yml (hunk 1 above)
Comment (RMcVelia) [reply .github/workflows/database-backup.yml:142]: We ran some tests a while back, and you didn't get much improvement on disk saved vs time taken when you increased the compression level. It currently compresses to 276MB which is pretty good. Would be worth increasing the level if it was a lot bigger. It also turned out this step wasn't really the issue (sanitised backup creation), it was failing on the previous step when it tried to create a local postgres db with the original backup.

Comment (defong) [reply .github/workflows/database-backup.yml:142]: Ok

File: .github/workflows/database-backup.yml (hunk 2 above)
Comment (defong) [reply .github/workflows/database-backup.yml:108]: Ok maybe one for another time then

**Review by defong — APPROVED** [review]
Let see how this fares... :ship:

**Pull request merged**